src/fragments/lib/datastore/flutter/data-access/observe-update-snippet.mdx

```diff
@@ -0,0 +1,47 @@
+```dart
+class MyApp extends StatefulWidget {
+  const MyApp({Key? key}) : super(key: key);
+
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  StreamSubscription<QuerySnapshot<Post>>? _stream;
+
+  // A reference to the retrieved post
+  late Post _post;
+
+  // Initialize the Amplify libraries
+  ...
+
+  void observeQuery() {
+    _stream = Amplify.DataStore.observeQuery(
+      Post.classType,
+      where: Post.ID.eq('123')
+    ).listen((QuerySnapshot<Post> snapshot) {
+      setState(() {
+        _post = snapshot.items.first;
+      });
+    });
+  }
+
+  Future<void> updatePost(newTitle) async {
+      final updatedPost = _post.copyWith(id: _post.id, title: 'Updated Title');
```

Comment:
`copyWith` will preserve non-overwritten fields, so ID doesn't need to be manually included.

```suggestion
  Future<void> updatePost(String newTitle) async {
      final updatedPost = _post.copyWith(title: newTitle);
```

Reply:
Thanks again!